lib/katello/tasks/reset.rake

```diff
@@ -1,29 +1,46 @@
 namespace :katello do
   namespace :reset_backends do
-    service_stop = "sudo /sbin/service %s status > /dev/null && sudo /sbin/service %s stop"
-    service_start = "sudo /sbin/service %s start"
+    service_stop = "sudo /usr/bin/systemctl status %s > /dev/null && sudo /usr/bin/systemctl stop %s"
+    service_start = "sudo /usr/bin/systemctl start %s"
 
-    task :pulp do
-      SERVICES = %w(httpd pulp_workers pulp_resource_manager pulp_celerybeat).freeze
-      system(service_stop.gsub("%s", "mongod"))
+    task :pulp_legacy do
+      SERVICES = %w(httpd pulp_workers pulp_resource_manager pulp_celerybeat squid qpidd pulp_streamer).freeze
 
       SERVICES.each { |s| system(service_stop.gsub("%s", s)) }
-      system("sudo rm -rf /var/lib/mongodb/pulp_database*")
-      system(service_start.gsub("%s", "mongod"))
-      sleep(10)
-      fail "Cannot migrate pulp database" unless system("sudo -u apache /usr/bin/pulp-manage-db")
+      system(service_start.gsub("%s", 'rh-mongodb34-mongod'))
+      system("mongo pulp_database --eval 'db.dropDatabase();'")
+      fail 'Cannot migrate Pulp database' unless system('sudo -u apache /usr/bin/pulp-manage-db')
 
       SERVICES.each { |s| system(service_start.gsub("%s", s)) }
-      puts "Pulp database reset."
+      puts 'Pulp2 database reset.'
     end
 
-    task :candlepin do
-      tomcat = File.exist?('/var/lib/tomcat') ? 'tomcat' : 'tomcat6'
+    task :pulp do
+      PULP_DIR = '/usr/local/lib/pulp/bin'.freeze
+
+      # Pull active workers from redis to see how many we need to start up
+      worker_count = `redis-cli --scan --pattern "rq:workers:reserved-resource-worker*" | wc -l`.chomp.to_i
+
+      services = %w(redis pulpcore-api pulpcore-resource-manager pulpcore-content)
+      while worker_count > 0
+        services.push("pulpcore-worker@#{worker_count}")
+        worker_count -= worker_count
+      end
 
+      services.each { |s| system(service_stop.gsub("%s", s)) }
+      system("runuser - postgres -c 'dropdb pulp -p 7878'")
+      fail 'Cannot migrate Pulp database' unless system("#{PULP_DIR}/django-admin migrate auth --no-input")
+      fail 'Cannot migrate Pulp database' unless system("#{PULP_DIR}/django-admin migrate --no-input")
+
+      services.each { |s| system(service_start.gsub("%s", s)) }
+      puts 'Pulp3 database reset.'
+    end
+
+    task :candlepin do
       system(service_stop.gsub("%s", tomcat))
```

Comment:
I think you're missing 
```ruby
tomcat = File.exist?('/var/lib/tomcat') ? 'tomcat' : 'tomcat6'
```

I hit this: 
```
NameError: undefined local variable or method `tomcat' for main:Object
/home/vagrant/katello/lib/katello/tasks/reset.rake:40:in `block (3 levels) in <top (required)>'
...
```

Reply:
Fixed, forgot to quote as string since I removed the variable